### PR TITLE
Implementing Spaces

### PR DESCRIPTION
**Pull request by shreyasajj**
I think have implemented spaces correctly, I have done some decent testing but let me know if it works for you

### CODE REVIEW

File: example-config.yaml [hunk 1]
```diff
@@ -57,6 +57,15 @@ bridge:
     # {{.}} is replaced with the contact list name (if available) or username (phone number or email) of the iMessage user.
     displayname_template: "{{.}} (iMessage)"
 
+    # Should the bridge create a space for each logged-in user and add bridged rooms to it?
+    personal_filtering_spaces: 
+        # Whether or not spaces should be enabled
+        enable: false
+        # The name of the space
+        name: "IMessage"
+        # The Image of the space
```
Comment (sumnerevans) [inline example-config.yaml:66]:
```suggestion
        # The MXC URI of the space
```

File: user.go [hunk 2]
```diff
@@ -117,3 +121,84 @@ func (user *User) UpdateDirectChats(chats map[id.UserID][]id.RoomID) {
...
+	return
+}
+
+func (user *User) GetSpaceRoom() id.RoomID {
+	if !user.bridge.Config.Bridge.PersonalFilteringSpaces.Enable {
+		return ""
+	}
+
+	if len(user.SpaceRoom) == 0 {
+
```
Comment (sumnerevans) [inline user.go:167]:
```suggestion
```

File: portal.go [hunk 3]
```diff
@@ -655,6 +659,7 @@ func (portal *Portal) CreateMatrixRoom(chatInfo *imessage.ChatInfo) error {
 		portal.bridge.user.UpdateDirectChats(map[id.UserID][]id.RoomID{puppet.MXID: {portal.MXID}})
 	}
 	if portal.bridge.user.DoublePuppetIntent != nil {
+
```
Comment (sumnerevans) [inline portal.go:662]:
```suggestion
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]
I haven't had a chance to review in-depth, but at first glance, it looks sane.

**Comment by shreyasajj** [comment]
I didn't hear anything back in a while and I wanted to check in. Did something go wrong, I would love to help you fix it

**Comment by shreyasajj** [comment]
There has been one problem I have noticed so far is that it says it changes it avatar image 4 times in a row for the imessage bridge bot. Not sure if that related to this or another problem but really just a minor inconvenience

**Pull request closed without merging**